src/components/Button/Button.tsx

```diff
@@ -0,0 +1,42 @@
+import classNames from 'classnames';
+import React from 'react';
+import Loader from 'components/Loader';
+import Text from 'components/Text';
+
+import styles from './Button.module.scss';
+
+export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  /** Состояние загрузки */
+  loading?: boolean;
+  /** Текст кнопки */
+  children: React.ReactNode;
+};
+
+// eslint-disable-next-line react/prop-types
+const Button: React.FC<ButtonProps> = ({  children, className, loading, disabled, onClick, ...props}) => {
+  const buttonClasses = classNames([
+    styles.button,
+    className,
+    disabled && styles.__disabled,
+    loading && styles.__loading,
+  ]);
+
+  const onClickWrapper = onClick ? (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!disabled && !loading && onClick) {
+      onClick(event);
+    }
+  } : undefined;
+
```

Comment:
можно немного упростить себе жизнь и проверять `onClick` перед вызовом: 
```suggestion
  const onClickWrapper = (event: React.MouseEvent<HTMLButtonElement>) => {
    if (!disabled && !loading) {
      onClick?.(event);
    }
  };
```
кроме того можно прямо в стилях для классов __disabled и __loading задать правило `pointer-events: none`, которое запретит клики, сделает курсор стандартным и позволит сделать обработчик еще проще, убрав проверки этих состояний из него.